test/unit/reports/published_attachments_report_test.rb

```diff
@@ -0,0 +1,35 @@
+require "test_helper"
+
+class Reports::PublishedAttachmentsReportTest < ActiveSupport::TestCase
+  test "returns a report containing file attachments" do
+    create(:published_detailed_guide, :with_file_attachment)
+
+    path = nil
+    Timecop.freeze do
+      path = Rails.root.join("tmp/attachments_#{Time.zone.now.strftime('%d-%m-%Y_%H-%M')}.csv")
+
+      Reports::PublishedAttachmentsReport.new.report
+    end
+
+    assert_equal Reports::PublishedAttachmentsReport::CSV_HEADERS, CSV.read(path)[0]
+    assert_equal 1, CSV.read(path, headers: true).count
+
+    File.delete(path) if File.exist?(path)
+  end
+
+  test "returns blank report if there are no eligible file attachments" do
+    create(:publication, :with_file_attachment)
```

Comment:
So a publication can't have a file attachment? 🤔 Or it can, but we just don't want it included in the report?

Reply:
It can but it won't get returned in the report unless the document is published. I've matched the factories document types (both now detailed guides) which should make the difference in test setup more apparent.

Reply:
Makes sense, thanks 👍